### ISSUE TITLE
Hi, is there visualize code to show the camera poses and 3DGS model at the same time like the paper?

### ISSUE THREAD

**Issue by JinXJinLeX** [mbortolon97/6dgs#17]
Thank you for your excellent work! I executed the relevant `train.py` and `pretrain_eval_attention.py`, but I don't know the meaning of the output and how to view the results. 
`python pretrain_eval_attention.py --exp_path ./output/ --out_path results.json --data_type tankstemple
Object 0d164aa9-c of category tt_Barn skipped because no valid checkpoint found
Object 34059a23-8 of category tt_Barn skipped because no valid checkpoint found
Object 7eb33c78-2 of category tt_Barn skipped because no valid checkpoint found
Object a5720f72-e of category tt_Barn skipped because no valid checkpoint found
Object 508cd869-4 of category tt_Caterpillar skipped because no valid checkpoint found
Object 68c8604f-a of category tt_Caterpillar skipped because no valid checkpoint found
Object 7fc3c673-9 of category tt_Caterpillar skipped because no valid checkpoint found
Object a22a6137-c of category tt_Caterpillar skipped because no valid checkpoint found
Object 034d668b-4 of category tt_Family skipped because no valid checkpoint found
Object 5f062846-7 of category tt_Family skipped because no valid checkpoint found
Object 8b47b72d-c of category tt_Family skipped because no valid checkpoint found
Object 1931fe8e-d of category tt_Ignatius skipped because no valid checkpoint found
Object 713c2b25-6 of category tt_Ignatius skipped because no valid checkpoint found
Object ee25457a-1 of category tt_Ignatius skipped because no valid checkpoint found
data_path:  /home/seu/xjl_work_space/3DGSLoc/6DGS/6dgs-main/data/datasets/TanksAndTemple/Barn
Found intrinsics.txt file, assuming Tanks And Temple data set!
Reading Training Transforms
336it [00:27, 12.36it/s]
Reading Test Transforms
48it [00:03, 12.34it/s]
Downloading: "https://github.com/facebookresearch/dinov2/zipball/main" to /home/seu/.cache/torch/hub/main.zip
/home/seu/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/swiglu_ffn.py:51: UserWarning: xFormers is not available (SwiGLU)
  warnings.warn("xFormers is not available (SwiGLU)")
/home/seu/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/attention.py:33: UserWarning: xFormers is not available (Attention)
  warnings.warn("xFormers is not available (Attention)")
/home/seu/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/block.py:40: UserWarning: xFormers is not available (Block)
  warnings.warn("xFormers is not available (Block)")
Downloading: "https://dl.fbaipublicfiles.com/dinov2/dinov2_vits14/dinov2_vits14_pretrain.pth" to /home/seu/.cache/torch/hub/checkpoints/dinov2_vits14_pretrain.pth
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 84.2M/84.2M [00:06<00:00, 13.0MB/s]
[19] loss: 0.17577396647162685
Eval on train...
0it [00:00, ?it/s]/home/seu/xjl_work_space/3DGSLoc/6DGS/6dgs-main/pose_estimation/line_intersection.py:237: UserWarning: Using torch.cross without specifying the dim arg is deprecated.
Please either pass the dim explicitly or simply use torch.linalg.cross.
The default value of dim will change to agree with that of linalg.cross in a future release. (Triggered internally at /opt/conda/conda-bld/pytorch_1704987280714/work/aten/src/ATen/native/Cross.cpp:63.)
  xaxis = torch.cross(up, direction)
74it [00:01, 40.88it/s]wrong c2w
79it [00:01, 40.80it/s]wrong c2w
336it [00:08, 40.60it/s]
Average loss score:  4.590764120510335e-06
Average Recall:  0.0
Time per element:  0.02463109294573466
Translation Error:  0.07245571295511798
Angular Error:  23.534449491472472
Smallest translation error:  164
Eval on validation...
48it [00:01, 39.88it/s]
Average loss score:  4.879735871080963e-06
Average Recall:  0.0
Time per element:  0.025076573093732197
Translation Error:  0.06387336005961212
Angular Error:  24.02740142742793
Smallest translation error:  28
[39] loss: 0.008731310997961827
Eval on train...
215it [00:05, 42.18it/s]wrong c2w
336it [00:08, 41.94it/s]
Average loss score:  4.588398319410339e-06
Average Recall:  0.0
Time per element:  0.023845869160833814
Translation Error:  0.07349011511541903
Angular Error:  20.564353545152006
Smallest translation error:  224
Eval on validation...
48it [00:01, 41.40it/s]
Average loss score:  4.662744011341147e-06
Average Recall:  0.0
Time per element:  0.024158760905265808
Translation Error:  0.07661747836391442
Angular Error:  25.615034480889637
Smallest translation error:  34
[59] loss: 0.0019696912591200544
Eval on train...
336it [00:08, 38.82it/s]
Average loss score:  3.7657720783372607e-06
Average Recall:  0.001101190476190476
Time per element:  0.02575850628671192
Translation Error:  0.0725383314219514
Angular Error:  20.830416210350535
Smallest translation error:  101
Eval on validation...
48it [00:01, 38.09it/s]
Average loss score:  3.939762269311586e-06
Average Recall:  0.0016666666666666666
Time per element:  0.026253779729207356
Translation Error:  0.06442635875040044
Angular Error:  25.543951238195103
Smallest translation error:  17
[79] loss: 0.0009780022389520582
Eval on train...
336it [00:08, 38.47it/s]
Average loss score:  2.7292113777832583e-06
Average Recall:  2.9761904761904762e-05
Time per element:  0.02599271635214488
Translation Error:  0.07163339216467215
Angular Error:  23.076330904449737
Smallest translation error:  271
Eval on validation...
16it [00:00, 38.95it/s]wrong c2w
48it [00:01, 38.95it/s]
Average loss score:  3.0198838132378114e-06
Average Recall:  0.0
Time per element:  0.025676916042963665
Translation Error:  0.12591456376685528
Angular Error:  30.353035112222035
Smallest translation error:  28
[99] loss: 0.0011009225247761377
Eval on train...
336it [00:08, 38.35it/s]
Average loss score:  2.139181192311363e-06
Average Recall:  0.0
Time per element:  0.026072890276000613
Translation Error:  0.06122003490842568
Angular Error:  21.22301219119912
Smallest translation error:  163
Eval on validation...
48it [00:01, 37.87it/s]
Average loss score:  2.618641001816968e-06
Average Recall:  0.0
Time per element:  0.02640780806541443
Translation Error:  0.06908532305775832
Angular Error:  24.347685620188713
Smallest translation error:  18
[119] loss: 0.0015979544891479235
Eval on train...
150it [00:03, 40.91it/s]wrong c2w
305it [00:07, 41.24it/s]wrong c2w
325it [00:07, 42.18it/s]wrong c2w
336it [00:08, 41.52it/s]
Average loss score:  3.1050029041880734e-06
Average Recall:  0.0004464285714285714
Time per element:  0.0240830211412339
Translation Error:  0.08569784139953776
Angular Error:  21.518922398842516
Smallest translation error:  82
Eval on validation...
48it [00:01, 41.72it/s]
Average loss score:  3.342802367948631e-06
Average Recall:  0.0
Time per element:  0.023974373936653137
Translation Error:  0.05449005154271921
Angular Error:  22.878454864025116
Smallest translation error:  17
[139] loss: 0.0009834378765390284
Eval on train...
336it [00:08, 40.66it/s]
Average loss score:  2.6976879334416114e-06
Average Recall:  0.0
Time per element:  0.024592116475105286
Translation Error:  0.08586044442061047
Angular Error:  18.67068262256327
Smallest translation error:  277
Eval on validation...
48it [00:01, 40.82it/s]
Average loss score:  3.140725932363845e-06
Average Recall:  0.0
Time per element:  0.024497538805007935
Translation Error:  0.08010060320763539
Angular Error:  21.14385896921158
Smallest translation error:  18
[159] loss: 0.0003683379875269566
Eval on train...
336it [00:09, 36.12it/s]
Average loss score:  1.7649091557103702e-06
Average Recall:  0.0010416666666666667
Time per element:  0.0276857784816197
Translation Error:  0.05506372120830098
Angular Error:  21.00961199278633
Smallest translation error:  136
Eval on validation...
48it [00:01, 36.07it/s]
Average loss score:  2.1336678794151944e-06
Average Recall:  0.0010416666666666667
Time per element:  0.027726401885350544
Translation Error:  0.0463666918706925
Angular Error:  21.40006493528684
Smallest translation error:  23
[179] loss: 0.001048185546710556
Eval on train...
76it [00:01, 39.36it/s]wrong c2w
176it [00:04, 39.50it/s]wrong c2w
336it [00:08, 39.30it/s]
Average loss score:  2.191203002409127e-06
Average Recall:  8.928571428571429e-05
Time per element:  0.025446964161736623
Translation Error:  0.08127178125626718
Angular Error:  21.497137639848958
Smallest translation error:  183
Eval on validation...
48it [00:01, 39.15it/s]
Average loss score:  2.7543169037376933e-06
Average Recall:  0.0
Time per element:  0.025543933113416035
Translation Error:  0.06483551163303976
Angular Error:  23.156466325124104
Smallest translation error:  15
[199] loss: 0.00059315841622265
Eval on train...
290it [00:07, 41.24it/s]wrong c2w
336it [00:08, 40.89it/s]
Average loss score:  2.216631959803922e-06
Average Recall:  0.0
Time per element:  0.02445759730679648
Translation Error:  0.07793376559857279
Angular Error:  22.460118793660687
Smallest translation error:  54
Eval on validation...
48it [00:01, 40.43it/s]
Average loss score:  2.6653517591531304e-06
Average Recall:  0.0
Time per element:  0.024737690885861714
Translation Error:  0.07030562633493294
Angular Error:  24.482368674129248
Smallest translation error:  18
[219] loss: 0.0006124759023350635
Eval on train...
280it [00:06, 42.25it/s]wrong c2w
336it [00:07, 42.15it/s]
Average loss score:  2.2512614718449086e-06
Average Recall:  0.0003869047619047619
Time per element:  0.023727319070271084
Translation Error:  0.07813919022079235
Angular Error:  20.79214023834183
Smallest translation error:  172
Eval on validation...
48it [00:01, 42.03it/s]
Average loss score:  2.8142644404501502e-06
Average Recall:  0.0
Time per element:  0.023796310027440388
Translation Error:  0.07111055694986135
Angular Error:  20.1152214979132
Smallest translation error:  3
[239] loss: 0.0005114811070370707
Eval on train...
4it [00:00, 38.24it/s]wrong c2w
336it [00:08, 39.36it/s]
Average loss score:  nan
Average Recall:  0.00014880952380952382
Time per element:  0.0254073760339192
Translation Error:  0.07175329393413406
Angular Error:  20.44410746260768
Smallest translation error:  80
Eval on validation...
48it [00:01, 39.29it/s]
Average loss score:  2.173495779800305e-06
Average Recall:  0.00020833333333333335
Time per element:  0.025456910332043965
Translation Error:  0.057065465337169975
Angular Error:  22.86979775627454
Smallest translation error:  28
[259] loss: 0.0003433374682117574
Eval on train...
336it [00:07, 42.57it/s]
Average loss score:  2.00749454917638e-06
Average Recall:  0.0038095238095238095
Time per element:  0.023493807940256028
Translation Error:  0.0635639738923471
Angular Error:  21.728751306732494
Smallest translation error:  69
`
